dup.go

```diff
@@ -0,0 +1,12 @@
+// +build !arm64
+// +build !windows
```

Comment:
These build directives should probably should be comparable to [this](https://github.com/golang/go/commit/ab7e82ef3b773468f008df439568e792eb24d5fd#diff-16d2d9fe3f2e73526d5ee31a5c87b928R5) snippet since `dup3` only exists on linux.